src/foam/nanos/crunch/ruler/AuthorizeUCJStatusOnPut.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch.ruler',
+  name: 'AuthorizeUCJStatusOnPut',
+  implements: [ 'foam.nanos.ruler.RuleAction' ],
+  tags: [ 'security' ],
+
+  javaImports: [
+    'foam.core.ContextAgent',
+    'foam.core.X',
+    'foam.nanos.auth.AuthService',
+    'foam.nanos.crunch.Capability',
+    'foam.nanos.crunch.UserCapabilityJunction',
+    'java.util.HashMap',
+    'static foam.nanos.crunch.CapabilityJunctionStatus.*'
+  ],
+
+  documentation: `
+    This rule prevents a UCJ status from being set above ACTION_REQUIRED if the
+    user in context does not have
```

Comment:
Do you need to finish this sentence?

Reply:
Weird... it happened again. I think I have an editor shortcut I'm not aware of that does this :/